Feature/global ens drivers modulefiles

StevenSimon-NOAA opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
All global_ens atmos and waves prep, stats, and plots processing has been tested interactively and confirmed.
- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Run all atmos and waves prep, stats, and plots drivers.
- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**
Yes.
- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**
Yes.
- [ ] Please complete this pull request review by **[Fill in date]**.</br>
ASAP.
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

Overview: This PR addresses module files and their sourcing in the global_ens driver scripts at each step. All driver scripts have been renamed to feature the component global_ens. PNG plots image names were corrected to feature "_" in place of ".". Added $DATA to the ush scripts. Pulled COMIN and COMOUT definitions from the scripts step to the jobs step.

StevenSimon-NOAA committed:
Introduce new module files and revised global_ens wave driver scripts to
feature/global_ens-drivers-modulefiles.

StevenSimon-NOAA committed:
Modify COMOUT in global_ens wave plotting driver script.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Stats step for CAM radar and severe vx

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update meso stats, driver scripts, and module files.

StevenSimon-NOAA committed:
Modify global_ens atmo stats and plots driver scripts to source
modulefiles.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Updating machine.conf to fix the bin dir and verbosities.

StevenSimon-NOAA committed:
Update global_ens module files with additional modules.

StevenSimon-NOAA committed:
Modify global_ens atmos driver scripts to include sourcing of module
files.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Feature/subseasonal missing data run capability.

StevenSimon-NOAA committed:
Modify global_ens NAEFS prep driver script.

StevenSimon-NOAA committed:
Update HOMEevs and COMIN paths in global_ens atmo driver scripts and
modify plot png names in plotting scripts.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
feature/global_det_atmos_config_fhr_list, Feature/drivers modules, 06/29
CRITICAL MESSAGE: EVS code update, Bugfix/cam plotting memory issue, and
Create ecflow suite and run NARRE components through it.

StevenSimon-NOAA committed:
Modified exevs_global_ens_atmos_gefs_precip_spatial_plots.sh by
removing buk from file paths.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Remove icing potential from wafs, and update code to use modulefiles to
load modules.

StevenSimon-NOAA committed:
Modified COMIN and COMOUT variables in global_ens_headline prep scipts, job, and driver. Updated global_ens_atmos_naefs_precip_plots script.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Feature/mesoscale plotting pcp sno modifications.

StevenSimon-NOAA committed:
Update global_ens module file at plots step with imagemagik module and
edit global_ens_atmos_gefs_precip_spatial_plots script to generate plot
gifs.

StevenSimon-NOAA committed:
Modify COMIN and COMOUT variables in gefs and naefs global_ens atmo
drivers, jobs, and scripts.

StevenSimon-NOAA committed:
Renamed jevs_global_ens_atmos_prep.sh and jevs_naefs_atmos_prep.sh.
Deleted jevs_global_ens_prep.sh. Added $DATA before run* script in
/ush/global_ens/evs_global_ens_headline_grid2grid.sh.

StevenSimon-NOAA committed:
Remove commented out COMMOUTs in exevs_global_ens_gefs_atmos_prep.sh.
Deleted exevs_global_ens_prep.sh.

StevenSimon-NOAA committed:
Update exevs_global_ens_gefs_atmos_prep.sh with component and run env
variable substitutions in ush calls. Rename evs_get_gens_atmos_data.sh
and evs_gens_atmos_g2g_prep_naefs.sh (gens->global_ens).

StevenSimon-NOAA committed:
Add $DATA before run* script in ush scripts.

StevenSimon-NOAA committed:
Undo ush global_ens renaming from 7/13/2023 for
evs_gens_atmos_g2g_prep_naefs.sh and evs_get_gens_atmos_data.sh. Revert
second component in exevs_global_ens_gefs_atmos_prep.sh paths back to
gens.

StevenSimon-NOAA committed:
Fix typo in  exevs_global_ens_gefs_atmos_prep.sh.

StevenSimon-NOAA committed:
Renamed exevs_global_ens_headline_prep.sh ->
exevs_global_ens_gefs_headline_prep.sh.

StevenSimon-NOAA committed:
Modify HOMEevs and COMIN env variables in global_ens atmos stats drivers
for cmce and ecme.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
feature/href.ve: update plot scripts to modify the plot names (replace "." by " _" ).
revert settings that should now be default in machine.conf.

StevenSimon-NOAA committed:
Modify HOMEevs and COMIN in global_ens gefs and naefs atmos driver scripts.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update Feature/subseasonal modulefiles. Update feature/global_det_atmos_fix_ccpa_map.

StevenSimon-NOAA committed:
Updated ush/global_ens/evs_get_gens_atmos_data.sh to get ccpa24hr file
for atmos.

StevenSimon-NOAA committed:
Modify HOMEevs and COMIN env variables in global_ens atmos plot driver
scripts.

StevenSimon-NOAA committed:
Remove explicit evs_ver env variable and rename prep driver scripts for
global_ens component.

StevenSimon-NOAA committed:
Rename global_ens stats and plots driver script names and PBS N entries.

StevenSimon-NOAA committed:
Edit PBS N names for jevs_global_ens_grid2grid_past90days_plots.sh and
jevs_global_ens_grid2obs_past31days_init_separate_plots.sh.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Updating analyses with the modulefile and also adding in machine.conf
ites.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/narre.ve.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/cam ptype.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/cam grid2grid.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/stats meso3. Update stats and plots driver/scripts for mesoscale component.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Implement feature/global_det_wave_rework.

StevenSimon-NOAA committed:
Update PBS -l place setting for global_ens plots driver scripts.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Remove J-JOB JEVS_GLOBAL_DET_ATMOS_HEADLINE_PLOTS.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Updating rtofs to include modulefile and remove touch from prep.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/cam headline.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/nfcens modulefiles.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Updating rtofs osisaf location in prep step.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/headplot aqm.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/href.ve. The driver script file names have been changed.

StevenSimon-NOAA committed:
Create individual ETS and FBIAS plots. Create individual plots for FSS
neighborhoods. Create individual plots for thresholds. Rename the
graphics to follow the EVS naming conventions.

StevenSimon-NOAA committed:
Remove redundant #!/bin/bash and export MET_bin_exec=bin from global
ensemble driver scripts at all steps. Increase memory of last 31 days profile 3 plot driver to 400 GB.

ShelleyMelchior-NOAA commented on dev/drivers/scripts/global_ens/plots/jevs_global_ens_wave_grid2obs_plots.sh:
```diff
@@ -13,37 +13,32 @@ set -x
-export evs_ver=$evs_ver
+#export evs_ver=$evs_ver
 export model_ver=$gefs_ver
 export obsproc_ver=$obsproc_ver
```
You can eliminate line 24: `#export evs_ver=$evs_ver`
You can also eliminate line 26: `export obsproc_ver=$obsproc_ver`. $obsproc_ver is set in run.ver

StevenSimon-NOAA replied:
Done! But do you want this to extend to all instances within all wave drivers?

StevenSimon-NOAA replied:
Applied changes to all wave drivers (prep, stats, and plots).

ShelleyMelchior-NOAA commented on dev/drivers/scripts/global_ens/prep/jevs_global_ens_naefs_atmos_prep.sh:
```diff
@@ -12,45 +10,33 @@
 export RUN=atmos
 export STEP=prep
-export COMPONENT=naefs
+#export COMPONENT=naefs
```
Remove line 20: `#export COMPONENT=naefs`

StevenSimon-NOAA replied:
Done!

ShelleyMelchior-NOAA commented on dev/drivers/scripts/global_ens/prep/jevs_global_ens_naefs_atmos_prep.sh:
```diff
@@ -12,45 +10,33 @@
+
+module reset
+
+source $HOMEevs/modulefiles/global_ens/global_ens_${STEP}.sh
```
Modify to use $COMPONENT in place of global_ens.

ShelleyMelchior-NOAA replied:
YOu can still take this one further ... use $COMPONENT for all global_ens instances.
From
`source $HOMEevs/modulefiles/${COMPONENT}/global_ens_${STEP}.sh`
 to
`source $HOMEevs/modulefiles/${COMPONENT}/${COMPONENT}_${STEP}.sh`.

StevenSimon-NOAA replied:
Done!

ShelleyMelchior-NOAA commented on scripts/global_ens/prep/exevs_global_ens_naefs_atmos_prep.sh:
```diff
@@ -19,10 +19,10 @@ export run_mpi=${run_mpi:-'yes'}
+#export COMOUT_gefs_bc=${COMOUT}.${INITDATE}/gefs_bc
+#export COMOUT_cmce_bc=${COMOUT}.${INITDATE}/cmce_bc
+#mkdir -p $COMOUT_gefs_bc
+#mkdir -p $COMOUT_cmce_bc
```
Let's remove lines 22-25 now that they are loaded in the prep J-job.

StevenSimon-NOAA replied:
Done!

StevenSimon-NOAA committed:
Eliminated redundant and commented out env variables and switched COMIN
in the prep drivers to $USER.

StevenSimon-NOAA committed:
Make all global_ens explicts to $COMPONENT.

ShelleyMelchior-NOAA commented:
I have started running the 4 prep drivers.

ShelleyMelchior-NOAA commented:
The 4 prep jobs have completed. You can view the log files here:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_headline_prep.o74133440`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_naefs_atmos_prep.o74134129`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_prep.o74134472`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_prep.o74132674`
And COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens`

All expected files seem to be produced in my test when comparing against emc.vpppg, with the excpetion of gefs_bc/. When I compare the number of files in the COMOUT dir, my test has fewer than emc.vpppg.
`$ ls /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens/atmos.20230729/gefs_bc | wc -l` ... 2,520                                           
vs
`$ ls /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens/atmos.20230729/gefs_bc | wc -l` ... 3,780

StevenSimon-NOAA commented:
> The 4 prep jobs have completed. You can view the log files here: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_headline_prep.o74133440` `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_naefs_atmos_prep.o74134129` `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_prep.o74134472` `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_prep.o74132674` And COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens`
> 
> All expected files seem to be produced in my test when comparing against emc.vpppg, with the excpetion of gefs_bc/. When I compare the number of files in the COMOUT dir, my test has fewer than emc.vpppg. `$ ls /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens/atmos.20230729/gefs_bc | wc -l` ... 2,520 vs `$ ls /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/global_ens/atmos.20230729/gefs_bc | wc -l` ... 3,780

That discrepancy for gefs_bc prep job output files is expected since my version is associated with naefs v6 (gefs_bc with 20 members) and the parallel's version is associated with naefs v7 (gefs_bc with 30 members). Going forward, it's not clear which version of naefs we should be opting for in the timeframe of this PR.

ShelleyMelchior-NOAA commented:
Thanks Steve. I think the EVS code should be in context of NAEFSv6 and therefore this PR is correct to have removed code for NAEFSv7.

I will move onto stats

AliciaBentley-NOAA commented:
Hi everyone,

You are correct. EVS needs to assess NCEPs operational models. NAEFSv6 is
the current operational version of NAEFS, not NAEFSv7. Thank you for making
that adjustment!

There will be a time, likely shortly after EVSv1 code delivery, when we
will have to update the EVSv1 code to verify NAEFSv7 instead. Steven Earle
recently got back from parental leave- we plan to ask him how he would like
to handle the NAEFS and WAFS upgrades happening in November very soon. We
will let you know what he says!

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬


On Tue, Aug 1, 2023 at 9:06 AM Shelley Melchior ***@***.***>
wrote:

> Thanks Steve. I think the EVS code should be in context of NAEFSv6 and
> therefore this PR is correct to have removed code for NAEFSv7.
>
> I will move onto stats
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1660274351>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUWOBEBQ3MRQITPV7KVEEDXTD5LZANCNFSM6AAAAAA2TFSXU4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

ShelleyMelchior-NOAA commented:
I am biting off the stats in chunks. I just completed all the runs for cmce stats.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/cmce.20230730`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/wmo.20230730/cmce`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/cmce`

logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_cmce_*.o*`

malloryprow commented on ush/global_ens/settings.py:
Not sure if these are related to other work but the files I sent to copy should have gone in /ush/global_ens/ush_gens_plot_py

ShelleyMelchior-NOAA replied:
Ohhh! There is a subdirectory that has the exact same filenames in it as the parent directory.

malloryprow replied:
There are. The wave component uses the ones in ush/global_ens and the atmos uses the ones in ush/global_ens/ush_gens_plot_py.

ShelleyMelchior-NOAA replied:
Thank you for that clarification, @malloryprow . I didn't pick up on the fact that your changes were in a lower subdirectory. @StevenSimon-NOAA is reverting the files that were erroneously copied into and will apply your changes to the atmos subdirectory. I wonder if it would be beneficial to use a different naming convention, or even just put atmos or wave where appropriate. I can see this triggering a problem again down the road.

ShelleyMelchior-NOAA replied:
@malloryprow , please confirm that the ush plotting files are as expected now.

malloryprow commented on ush/global_ens/lead_average.py:
Not sure if these are related to other work but the files I sent to copy should have gone in /ush/global_ens/ush_gens_plot_py

malloryprow commented on ush/global_ens/time_series.py:
Not sure if these are related to other work but the files I sent to copy should have gone in /ush/global_ens/ush_gens_plot_py

StevenSimon-NOAA committed:
Restore old python plotting ush files in /ush/global_ens.

ShelleyMelchior-NOAA commented:
I just completed all of the runs for ecme stats. Please review against the EVS para.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/ecme.20230730`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/ecme`

logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_ecme_atmos_*`

StevenSimon-NOAA committed:
Redo of copy transfer of lead_average.py, settings.py, and
time_series.py to ush/global_ens/ush_gens_plot_py subdirectory. Create
individual ETS and FBIAS plots. Create individual plots for FSS
neighborhoods. Create individual plots for thresholds. Rename the
graphics to follow the EVS naming conventions.

StevenSimon-NOAA committed:
Update $HOMEevs, $COMIN, and $COMOUT to eliminate gitworkspace directory
and emc.vpppg from their paths.

ShelleyMelchior-NOAA commented:
I just completed all of the runs for naefs stats. Please review against the EVS para.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/naefs.20230730`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/naefs`

logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_naefs_*` (the latest runs)

It looks like atmos_grid2obs and atmos_grid2grid terminated b/c they exceeded wallclock. I have extended walltime and am re-running. I will update the PR once g2o and g2g re-runs are complete.
headline and atmos_precip ran to completion

ShelleyMelchior-NOAA commented:
FYI -- re-runs of naefs grid2obs and grid2grid have completed. See output as directed in previous comment.

malloryprow commented:
Changes to the ush files are good!

StevenSimon-NOAA commented:
> I am biting off the stats in chunks. I just completed all the runs for cmce stats. COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/cmce.20230730` COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/wmo.20230730/cmce` COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/cmce`
> 
> logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_cmce_*.o*`

ShelleyMelchior-NOAA commented:
The naefs g2g and g2o stats re-runs had the walltime bumped out by 30 minutes for both cases. That was sufficient for both jobs to complete. However, if you are not having runtime problems in your cron processing or in the EVS parallel, I will chalk up my problems to the nature of the business of the dev machine at the time of day that I ran and a code change is not needed.

ShelleyMelchior-NOAA commented:
The gefs stats processing has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/gefs.20230730`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/gefs`
logfies: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_gefs_*`

The sst didn't run correctly b/c of a typo (my err) in the definition of the HOMEevs variable. I am re-running that one right now.

ShelleyMelchior-NOAA commented:
The gefs sst stats has completed. NOTE the new date b/c I am running a day later. Please review.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/gefs.20230731`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230731/gefs`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_gefs_atmos_sst_stats.o88078972`

StevenSimon-NOAA commented:
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/cmce

We are generating more stats files than the EVS parallel in COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/cmce. Is this a product of the dev machine transition?

StevenSimon-NOAA commented:
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/ecme

We are generating more stats files than the EVS parallel in COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/ecme. Is this a product of the dev machine transition?

StevenSimon-NOAA commented:
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/naefs

We are generating more stats files than the EVS parallel in COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/naefs. Is this a product of the dev machine transition?

StevenSimon-NOAA commented:
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/gefs

We are generating more stats files than the EVS parallel in COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230730/gefs for product except cnv. Is this a product of the dev machine transition?

StevenSimon-NOAA commented:
> The gefs sst stats has completed. NOTE the new date b/c I am running a day later. Please review. COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/gefs.20230731` COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/atmos.20230731/gefs` logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_gefs_atmos_sst_stats.o88078972`

All gefs sst stats output files match up with the EVS parallel for sst.

ShelleyMelchior-NOAA commented:
I think we can contribute the miscount in stats files to the production machine switch. My tests yield more stats files than the EVS para runs for the same VDATE likely b/c my tests ran later after more prep data files were transferred and available. NOTE: my stats processing pointed to emc.vpppg prep data files.

I will proceed w/ plots processing.

ShelleyMelchior-NOAA commented:
Let's do some clean up w/ this PR, too! Can you please remove any driver scripts that when you long list (ls -ltr) on them they show the size is 7 bytes. For example: 
`-rwxr-xr-x 1 shelley.melchior vpppg    7 Aug  1 15:04 jevs_global_ens_tropcyc_plots.sh`
I see 6 in plots/ and 6 in stats/.

StevenSimon-NOAA committed:
Remove placeholder driver scripts.

StevenSimon-NOAA commented:
> Let's do some clean up w/ this PR, too! Can you please remove any driver scripts that when you long list (ls -ltr) on them they show the size is 7 bytes. For example: `-rwxr-xr-x 1 shelley.melchior vpppg 7 Aug 1 15:04 jevs_global_ens_tropcyc_plots.sh` I see 6 in plots/ and 6 in stats/.

I just removed these placeholder driver scripts and pushed the updates to GitHub. It appears that the PR updated accordingly as well.

ShelleyMelchior-NOAA commented:
All plots processing has completed. But I am 2 tar files short in COMOUT (as compared to the EVS para COMOUT for the same VDATE).
atmos ...
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20230731`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_*plots*`
I had to re-run `jevs_global_ens_grid2obs_past90days_plots.sh` b/c of my typo for HOMEevs.
I had to re-run `jevs_global_ens_precip_spatial_plots.sh` b/c of walltime exceedance.
Now all expected tar files are available in COMOUT.

wave ...
COMOUT=` /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/global_ens/wave.20230801`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_wave_grid2obs_plots.o88190558`

Please untar the files and review the png graphics, both for correct filenames as well as reasonable content of the plots.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/radar_svr_plots and feature/global_det_atmos_prep_metfra.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/radar_svr_modulefiles.

StevenSimon-NOAA committed:
Update valid time from 00_12z to 00z_12z. Update cape.10 to cape_10.

StevenSimon-NOAA committed:
Modify performance diagram image names from "f12_to_f384" to "f384".

StevenSimon-NOAA committed:
Remove valid time from filenames of seperate init plots.

StevenSimon-NOAA committed:
Restored identation.

StevenSimon-NOAA committed:
Restored identation to original.

ShelleyMelchior-NOAA commented:
I will re-run the following plots drivers:
`jevs_global_ens_grid2obs_past31days_init_separate_plots.sh`
`jevs_global_ens_grid2obs_past90 days_init_separate_plots.sh`
`jevs_global_ens_sea_ice_past31days_plots.sh`
`jevs_global_ens_sea_ice_past90days_plots.sh`

StevenSimon-NOAA committed:
Update global_ens atmos plots scripts to include buk region formatting
for plots filenames.

ShelleyMelchior-NOAA commented:
Thanks @StevenSimon-NOAA ! I re-ran the 4 identified plots drivers. I scp'd the resultant tar files to emcrzdm so that they can be viewed while WCOSS2 is unavailable. I untar'd the files into the following locations:
[grid2obs_past31days_init_separate](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/)
[grid2obs_past90days_init_separate](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/)
[sea_ice_past31days](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/)
[sea_ice_past90days](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/)
Have a look! Thanks!

StevenSimon-NOAA commented:
> Thanks @StevenSimon-NOAA ! I re-ran the 4 identified plots drivers. I scp'd the resultant tar files to emcrzdm so that they can be viewed while WCOSS2 is unavailable. I untar'd the files into the following locations: [grid2obs_past31days_init_separate](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/) [grid2obs_past90days_init_separate](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/) [sea_ice_past31days](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/) [sea_ice_past90days](https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/) Have a look! Thanks!

It looks like the plots filenames are now formatted correctly. However, the plot quantities appear to run short again. Most notably for 31 day sea ice. There are no rmse_me time series plots generated here. But it appears that the EVS Parallel is also not generating these plots as well. Some of these plots have not been generated for a few days (our EVS version and the EVS Parallel). But it looks like the missing ones were last generated on July 31st. No changes were made to the global_ens drivers, jobs, scripts, and/or ush files. And we merged nothing new to the EVS Parallel. However The last 31 day log files indicate that FileFoundError issue again.

AliciaBentley-NOAA commented:
Hi Steve and Shelley,

Do we know why some jobs aren't completing? What do the log files say? I
suggested increasing the wallclock time in a previous email to make sure
that jobs complete, but perhaps it's a memory issue instead?

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬


On Sat, Aug 5, 2023 at 3:30 PM Steven Simon ***@***.***>
wrote:

> Thanks @StevenSimon-NOAA <https://github.com/StevenSimon-NOAA> ! I re-ran
> the 4 identified plots drivers. I scp'd the resultant tar files to emcrzdm
> so that they can be viewed while WCOSS2 is unavailable. I untar'd the files
> into the following locations: grid2obs_past31days_init_separate
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/>
> grid2obs_past90days_init_separate
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/>
> sea_ice_past31days
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/>
> sea_ice_past90days
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/>
> Have a look! Thanks!
>
> It looks like the plots filenames are now formatted correctly. However,
> the plot quantities appear to run short again. Most notably for 31 day sea
> ice. There are no rmse_spread time series plots generated here.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666587174>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

StevenSimon-NOAA commented:
> Hi Steve and Shelley, Do we know why some jobs aren't completing? What do the log files say? I suggested increasing the wallclock time in a previous email to make sure that jobs complete, but perhaps it's a memory issue instead? Take care, Alicia
> […](#)
> ___________________________________ *Alicia M. Bentley, Ph.D.* Global Verification Project Lead Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/> NOAA/NWS/NCEP/EMC (VPPPG Branch) Work Phone: ‪(301) 683-3783‬
> On Sat, Aug 5, 2023 at 3:30 PM Steven Simon ***@***.***> wrote: Thanks @StevenSimon-NOAA <https://github.com/StevenSimon-NOAA> ! I re-ran the 4 identified plots drivers. I scp'd the resultant tar files to emcrzdm so that they can be viewed while WCOSS2 is unavailable. I untar'd the files into the following locations: grid2obs_past31days_init_separate <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/> grid2obs_past90days_init_separate <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/> sea_ice_past31days <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/> sea_ice_past90days <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/> Have a look! Thanks! It looks like the plots filenames are now formatted correctly. However, the plot quantities appear to run short again. Most notably for 31 day sea ice. There are no rmse_spread time series plots generated here. — Reply to this email directly, view it on GitHub <[#171 (comment)](https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666587174)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4> . You are receiving this because you commented.Message ID: ***@***.***>

According to the logs for the EVS Parallel, it does not appear to be a walltime issue. The plots drivers do not require much time to run (30 minutes is more than enough for sea ice). The only errors I can identify in the logs are these FileFoundErrors. There may be memory issues at play, but there is no clear error message that I can identify through a / search.

AliciaBentley-NOAA commented:
Hi Steve,

Thanks. The dev supercomputer will be down Monday and most of
Tuesday. Would it be possible for you to copy and paste some of your log
files into a Google Doc for people to read on Monday? There must be a
reason why the jobs aren't making plots. Perhaps Mallory, Shelley, or Perry
can help you find the reason on Monday. I know that you and Mallory already
have a webpage meeting planned during that time.

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬


On Sun, Aug 6, 2023 at 10:29 AM Steven Simon ***@***.***>
wrote:

> Hi Steve and Shelley, Do we know why some jobs aren't completing? What do
> the log files say? I suggested increasing the wallclock time in a previous
> email to make sure that jobs complete, but perhaps it's a memory issue
> instead? Take care, Alicia
> … <#m_-4863627275540734992_>
> ___________________________________ *Alicia M. Bentley, Ph.D.* Global
> Verification Project Lead Model Evaluation Group (MEG)
> https://www.emc.ncep.noaa.gov/users/meg/home/ NOAA/NWS/NCEP/EMC (VPPPG
> Branch) Work Phone: ‪(301) 683-3783‬
> On Sat, Aug 5, 2023 at 3:30 PM Steven Simon *@*.*> wrote: Thanks
> @StevenSimon-NOAA <https://github.com/StevenSimon-NOAA>
> https://github.com/StevenSimon-NOAA <https://github.com/StevenSimon-NOAA> !
> I re-ran the 4 identified plots drivers. I scp'd the resultant tar files to
> emcrzdm so that they can be viewed while WCOSS2 is unavailable. I untar'd
> the files into the following locations: grid2obs_past31days_init_separate
> https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/>
> grid2obs_past90days_init_separate
> https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/>
> sea_ice_past31days
> https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/>
> sea_ice_past90days
> https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/
> <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/>
> Have a look! Thanks! It looks like the plots filenames are now formatted
> correctly. However, the plot quantities appear to run short again. Most
> notably for 31 day sea ice. There are no rmse_spread time series plots
> generated here. — Reply to this email directly, view it on GitHub <#171
> (comment)
> <https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666587174>>, or
> unsubscribe
> https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4
> <https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4>
> . You are receiving this because you commented.Message ID: @.*>
>
> According to the logs for the EVS Parallel, it does not appear to be a
> walltime issue. The plots drivers do not require much time to run (30
> minutes is more than enough for sea ice). The only errors I can identify in
> the logs are these FileFoundErrors. There may be memory issues at play, but
> there is no clear error message that I can identify through a / search.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666878944>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUWOBCQHGDFXTICKCG6NBLXT6S33ANCNFSM6AAAAAA2TFSXU4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

StevenSimon-NOAA commented:
> Hi Steve, Thanks. The dev supercomputer will be down Monday and most of Tuesday. Would it be possible for you to copy and paste some of your log files into a Google Doc for people to read on Monday? There must be a reason why the jobs aren't making plots. Perhaps Mallory, Shelley, or Perry can help you find the reason on Monday. I know that you and Mallory already have a webpage meeting planned during that time. Take care, Alicia
> […](#)
> ___________________________________ *Alicia M. Bentley, Ph.D.* Global Verification Project Lead Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/> NOAA/NWS/NCEP/EMC (VPPPG Branch) Work Phone: ‪(301) 683-3783‬ On Sun, Aug 6, 2023 at 10:29 AM Steven Simon ***@***.***> wrote:
> Hi Steve and Shelley, Do we know why some jobs aren't completing? What do the log files say? I suggested increasing the wallclock time in a previous email to make sure that jobs complete, but perhaps it's a memory issue instead? Take care, Alicia … <#m_-4863627275540734992_> ___________________________________ *Alicia M. Bentley, Ph.D.* Global Verification Project Lead Model Evaluation Group (MEG) https://www.emc.ncep.noaa.gov/users/meg/home/ NOAA/NWS/NCEP/EMC (VPPPG Branch) Work Phone: ‪(301) 683-3783‬ On Sat, Aug 5, 2023 at 3:30 PM Steven Simon *@*.*> wrote: Thanks @StevenSimon-NOAA <https://github.com/StevenSimon-NOAA> https://github.com/StevenSimon-NOAA <https://github.com/StevenSimon-NOAA> ! I re-ran the 4 identified plots drivers. I scp'd the resultant tar files to emcrzdm so that they can be viewed while WCOSS2 is unavailable. I untar'd the files into the following locations: grid2obs_past31days_init_separate https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/ <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o31/> grid2obs_past90days_init_separate https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/ <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/g2o90/> sea_ice_past31days https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/ <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice31/> sea_ice_past90days https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/ <https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/global_ens/SSi-PR/seaice90/> Have a look! Thanks! It looks like the plots filenames are now formatted correctly. However, the plot quantities appear to run short again. Most notably for 31 day sea ice. There are no rmse_spread time series plots generated here. — Reply to this email directly, view it on GitHub <#171 (comment) <[#171 (comment)](https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666587174)>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4 <https://github.com/notifications/unsubscribe-auth/APUWOBETRBHOOS5N5YWLJALXT2NLZANCNFSM6AAAAAA2TFSXU4> . You are receiving this because you commented.Message ID: @.*> According to the logs for the EVS Parallel, it does not appear to be a walltime issue. The plots drivers do not require much time to run (30 minutes is more than enough for sea ice). The only errors I can identify in the logs are these FileFoundErrors. There may be memory issues at play, but there is no clear error message that I can identify through a / search. — Reply to this email directly, view it on GitHub <[#171 (comment)](https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666878944)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APUWOBCQHGDFXTICKCG6NBLXT6S33ANCNFSM6AAAAAA2TFSXU4> . You are receiving this because you commented.Message ID: ***@***.***>

Unfortunately, vi will not let me paste anything I copy from it into Notepad or a Google Doc. And Tectia will not allow me to do a secure file transfer without errors. Here are the directories to the log files of interest in case that you have better luck copy and pasting them from vi: /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_sea_ice_past31days_plots.o88833794 and   /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_grid2obs_past31days_init_separate_plots.o88833767.

StevenSimon-NOAA commented:
Unfortunately, vi in Tectia will not allow me to paste anything I copy from it into a Google Doc or Notepad. I do not why that is the case. Also, Tectia will not allow me to successfully log into its Secure File Transfer Interface. Here are the relevant logfiles: /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_sea_ice_past31days_plots.o88833794 and/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/ jevs_global_ens_grid2obs_past31days_init_separate_plots.o88833767. You can find similar in the emc.vpppg directories for both cases. Again, the missing plots issue appears in the PR version and Parallel version.

AliciaBentley-NOAA commented:
Hi Steve,

Tectia does allow you to copy and paste into a Google doc. I'm sorry that
isn't working for you. Instead, I have pushed your log files to emcrzdm for
you to take a look at with others tomorrow while the dev machine is down.
You can find them here:
https://emc.ncep.noaa.gov/users/Alicia.Bentley/out_files/

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬


On Sun, Aug 6, 2023 at 4:25 PM Steven Simon ***@***.***>
wrote:

> Unfortunately, vi in Tectia will not allow me to paste anything I copy
> from it into a Google Doc or Notepad. I do not why that is the case. Also,
> Tectia will not allow me to successfully log into its Secure File Transfer
> Interface. Here are the relevant logfiles:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_sea_ice_past31days_plots.o88833794
> and/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/
> jevs_global_ens_grid2obs_past31days_init_separate_plots.o88833767.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/171#issuecomment-1666960419>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUWOBFYSSCUAOGVSGK4AHDXT74SBANCNFSM6AAAAAA2TFSXU4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

StevenSimon-NOAA commented:
Thank you Alicia. Again, I attempted several variations of the "yy" command in vi. The contents were copied. But each time I attempted to paste them into a blank Google Doc, nothing appeared in the Google Doc.

ShelleyMelchior-NOAA commented:
I think the yanking (yy) is specific to vi. All I knwo to do is to highlight the lines, then right click the mouse for copy. But those log files are probhibitively long to iteratively copy and paste into a google doc. Maybe @AliciaBentley-NOAA  knows the magic sauce ...

StevenSimon-NOAA commented:
> I think the yanking (yy) is specific to vi. All I knwo to do is to highlight the lines, then right click the mouse for copy. But those log files are probhibitively long to iteratively copy and paste into a google doc. Maybe @AliciaBentley-NOAA knows the magic sauce ...

I can copy and paste through a piecemeal approach from the interactive vi, using the mouse's right click copy and right click paste. But the interactive vi editor only allows so many lines at a time to be copied that way. I also attempted using that method after using "cat filename". That only displays a portion of the entire log file in the terminal window.

StevenSimon-NOAA committed:
Add mkdir -p  plot_dir to all applicable global_ens plots scripts.
Remove redundant or empty applicable global_ens plots scripts.

ShelleyMelchior-NOAA commented:
I see the changes for the plots scripts. I will re-run the whole suite of plots drivers and report back.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Update feature/wafs_atmos2.

StevenSimon-NOAA committed:
Add "all" to global_ens vertprof plots filenames to indicate that these
vertprof plots encompass all plevels.

ShelleyMelchior-NOAA commented:
Plots have been fully regenerated this morning. I made the changes to modulefiles/ locally, but will look for them to be formally pushed to the feature branch.
Please untar all of the tar files in COMOUT and review.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20230802`

The one thing I see is that the headline plots processing did not produce the tar file in its COMOUT space:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/global_ens/headline.20230802`
I cannot see straight away what the problem is. Please review the log file:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_headline_plots.o89141102`

StevenSimon-NOAA commented:
> jevs_global_ens_headline_plots.o89141102

Traceback (most recent call last):
  File "/lfs/h2/emc/stmp/shelley.melchior/evs/tmpnwprd/jevs_global_ens_headline_plots.89141102.dbqs01/evs_global_ens_headline_plot.py", line 58, in <module>
    gfs_scores[i] = text_file1.readline()
ValueError: could not convert string to float: ''
+ cp NH_H500_PAC_2023.png /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/global_ens/headline.20230802/evs.plot.gefs.headline.hgt_p500.20230101_20230802.nhem.png
cp: cannot stat 'NH_H500_PAC_2023.png': No such file or directory


This is the error.

StevenSimon-NOAA committed:
Update global_ens modulefiles at each step.

StevenSimon-NOAA committed:
Fix buk region abbreviations in naefs plots filenames.

ShelleyMelchior-NOAA commented:
I re-ran naefs precip and g2o for both 31 and 90 days. Please review.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20230802`
Note the later time stamp for the 4 new tar files that were generated.

StevenSimon-NOAA committed:
Update naefs g2o and precip plots filenames buk regions.

StevenSimon-NOAA committed:
Fix 250 mb pressure level naming convention in naefs atmos g2o plots
filenames.

StevenSimon-NOAA committed:
Merge branch 'develop' into feature/global_ens-drivers-modulefiles.
Implement feature/subseasonal script name changes. Update
feature/ve-evs-prod.

ShelleyMelchior-NOAA reviewed (approved):
The expected changes are approved. The PR may be merged.

There is still some outstanding work for the global_ens component that needs to be addressed prior to code delivery. A follow-on PR will be submitted.
